core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -1582,6 +1582,42 @@ class WorkspaceService(protected val userInfo: UserInfo,
       }
     }
 
+  def retrySubmission(workspaceName: WorkspaceName, submissionRetry: SubmissionRetry, submissionId: String): Future[RetriedSubmissionReport] = {
+    getWorkspaceContextAndPermissions(workspaceName, SamWorkspaceActions.write) flatMap { workspaceContext =>
+      dataSource.inTransaction { dataAccess =>
+        withSubmission(workspaceContext, submissionId, dataAccess) { submission =>
+          val newSubmission = Submission(submissionId = submissionId,
```

Comment:
I think this is creating a new submission with the same UUID as the old submission. I'm surprised this isn't failing any tests!